### PR TITLE
Add travis build file

### PR DESCRIPTION
**Pull request by yiheng**
This patch adds basic Travis build configuration file. Once this is merged, we can use Travis for our CI process.

Manually test it on Travis.

@TisonKun @gosonzhang Please help review, thanks

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+language: java
+
+jdk:
+  - oraclejdk8
+
+cache:
+  directories:
+    - $HOME/.m2
+
+install:
```
Comment (yiheng) [inline .travis.yml:32]: reference: https://docs.travis-ci.com/user/installing-dependencies/

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations under the License.
+#
+
+sudo: required
+dist: trusty
+language: java
+
+jdk:
+  - oraclejdk8
```
Comment (yiheng) [inline .travis.yml]: not sure if we should use openjdk or oraclejdk here

Comment (chenjunjiedada) [reply .travis.yml]: +1 for openjdk.

Comment (tisonkun) [reply .travis.yml]: +1 for openjdk

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
```
Comment (tisonkun) [inline .travis.yml]: IIUC we don't need shebang.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+sudo: required
+dist: trusty
+language: java
+
+jdk:
+  - oraclejdk8
+
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (tisonkun) [inline .travis.yml:30]: Maybe we should exclude our own artifacts by

```
before_cache:
  - rm -rf $HOME/.m2/repository/com/tencent/tubemq/
```

**Review by tisonkun — CHANGES_REQUESTED** [review]
Thanks for integrating CI support @yiheng. I left some inline comments.

File: .travis.yml (hunk 4 above)
Comment (yiheng) [reply .travis.yml:30]: Currently, we just `mvn verify`. But I think it's fine to add this line.

**Review by tisonkun — APPROVED** [review]
This change looks good to me now. Please manually re-test by triggering a travis job before merge.

**Comment by gosonzhang** [comment]
Thanks

File: .travis.yml (hunk 2 above)
Comment (kaynewu) [reply .travis.yml]: +1 for openjdk.

**Pull request merged**